Userland/Utilities/disasm.cpp

```diff
@@ -99,38 +127,69 @@ ErrorOr<int> serenity_main(Main::Arguments args)
+
+                instruction_symbols.appendff("{} ({}):\n", demangle(current_zero_size_symbol->name), current_zero_size_symbol->format_symbol_address());
+            }
+        }
+        // Handle ranged symbols separately.
+        if (current_ranged_symbol < ranged_symbols.end() && !current_ranged_symbol->contains(virtual_offset)) {
             if (!is_first_symbol && current_instruction_is_in_symbol) {
                 // The previous instruction was part of a symbol that doesn't cover the current instruction, so separate it from the current instruction with a newline.
-                outln();
-                current_instruction_is_in_symbol = (current_symbol + 1 < symbols.end() && (current_symbol + 1)->contains(virtual_offset));
+                needs_separator = true;
+                current_instruction_is_in_symbol = (current_ranged_symbol + 1 < ranged_symbols.end() && (current_ranged_symbol + 1)->contains(virtual_offset));
             }
 
-            // Try to find symbol covering current instruction, if one exists.
-            while (current_symbol + 1 < symbols.end() && !(current_symbol + 1)->contains(virtual_offset) && (current_symbol + 1)->address() <= virtual_offset) {
-                ++current_symbol;
+            // Print "dangling" symbols preceding the current instruction.
+            while (current_ranged_symbol + 1 < ranged_symbols.end() && !(current_ranged_symbol + 1)->contains(virtual_offset) && (current_ranged_symbol + 1)->address() <= virtual_offset) {
+                ++current_ranged_symbol;
                 if (!is_first_symbol)
-                    outln("\n({} ({:p}-{:p}))\n", demangle(current_symbol->name), current_symbol->address(), current_symbol->address_end());
+                    dangling_symbols.appendff("\n({} ({}))\n", demangle(current_ranged_symbol->name), current_ranged_symbol->format_symbol_address());
             }
-            while (current_symbol + 1 < symbols.end() && (current_symbol + 1)->contains(virtual_offset)) {
+            // Find and print all symbols covering the current instruction.
+            while (current_ranged_symbol + 1 < ranged_symbols.end() && (current_ranged_symbol + 1)->contains(virtual_offset)) {
                 if (!is_first_symbol && !current_instruction_is_in_symbol)
-                    outln();
-                ++current_symbol;
+                    needs_separator = true;
+                ++current_ranged_symbol;
                 current_instruction_is_in_symbol = true;
-                outln("{} ({:p}-{:p}):", demangle(current_symbol->name), current_symbol->address(), current_symbol->address_end());
+
+                instruction_symbols.appendff("{} ({}):\n", demangle(current_ranged_symbol->name), current_ranged_symbol->format_symbol_address());
             }
 
             is_first_symbol = false;
         }
 
+        if (needs_separator)
+            outln();
+        // Insert extra newline after the "dangling" symbols.
```

Comment:
What is this comment referring to? `dangling_symbols_text` includes an extra newline, but we don't insert any newlines here?

Reply:
Seems to refer to the lines of code above, will move it up.